soap/src/java/com/zimbra/soap/admin/message/GetAddressListInfoResponse.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * ***** BEGIN LICENSE BLOCK *****
+ * Zimbra Collaboration Suite Server
+ * Copyright (C) 2018 Synacor, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License as published by the Free Software Foundation,
+ * version 2 of the License.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ * You should have received a copy of the GNU General Public License along with this program.
+ * If not, see <http://www.gnu.org/licenses/>.
+ * ***** END LICENSE BLOCK *****
+ */
+package com.zimbra.soap.admin.message;
+
+import javax.xml.bind.annotation.XmlAccessType;
+import javax.xml.bind.annotation.XmlAccessorType;
+import javax.xml.bind.annotation.XmlAttribute;
+import javax.xml.bind.annotation.XmlElement;
+import javax.xml.bind.annotation.XmlRootElement;
+
+import com.zimbra.common.soap.AdminConstants;
+import com.zimbra.soap.admin.type.EntrySearchFilterInfo;
+import com.zimbra.soap.type.GalSearchType;
+
+/**
+ * @zm-api-command-auth-required true
+ * @zm-api-command-admin-auth-required true
+ * @zm-api-command-description get address list info response
+ */
+@XmlAccessorType(XmlAccessType.NONE)
+@XmlRootElement(name=AdminConstants.E_GET_ADDRESS_LIST_INFO_RESPONSE)
+public class GetAddressListInfoResponse {
+
+    /**
+     * @zm-api-field-tag id
+     * @zm-api-field-description zimbra id of the address list
+     */
+    @XmlAttribute(name=AdminConstants.E_ID /* id */, required=true)
+    private String id;
+
+    /**
+     * @zm-api-field-tag name
+     * @zm-api-field-description name of the address list
+     */
+    @XmlElement(name=AdminConstants.E_NAME /* name */, required=false)
+    private String name;
+
+    /**
+     * @zm-api-field-tag desc
+     * @zm-api-field-description description of the address list
+     */
+    @XmlElement(name=AdminConstants.E_DESC /* desc */, required=false)
+    private String desc;
+
+    /**
+     * @zm-api-field-tag searchFilter
+     * @zm-api-field-description search filter for conditions
+     */
+    @XmlElement(name=AdminConstants.E_SEARCH_FILTER /* searchFilter */, required=false)
+    private EntrySearchFilterInfo searchFilter;
+
+    /**
+     * @zm-api-field-tag type
+     * @zm-api-field-description gal search type
+     */
+    @XmlAttribute(name=AdminConstants.A_TYPE /* type */, required=false)
```

Comment:
required=true
address list will always have gal search type. while creating address list if type is not provided, "all" is set by default. so while fetching address list, type will always be there.